### ISSUE TITLE
Android Developers would like to understand how to map their knowledge into Flutter

### ISSUE THREAD

**Issue by eseidelGoogle** [flutter/flutter#8942]
One example came up in Gitter just now:
RecyclerView.Adapter => ListView.builder.

@mehmetf also mentioned once that he had some knowledge here (which he may even have written down).

analogs was the term used for the web version of this:
https://flutter.io/web-analogs/ (adding that for searchability)

**Comment by mehmetf** [comment]
tl;dr We don't really have anything written down.  At first, we tried to do this kind of systematic translation from our native code but we found that it was easier to learn proper patterns from Flutter Gallery and apply them to the app's design.

We wanted our app code to resemble what a Flutter app is supposed to be. For instance, our Android implementation is inheritance-heavy which is an anti-pattern in Flutter. Another example is our store system which uses EventBus for Android but a pub-sub system for Flutter which we found is more Flutter friendly. We also caught non-Material patterns in our design and pushed back on UX to fix these rather than creating workarounds in Flutter.

Due to these factors, I would avoid mapping terminology and patterns directly from Android/iOS to Flutter. I would much rather see a guideline  on how to build different components of the app:
- Model layer design.
- Lists, cards and other common ui patterns.

This also avoids the danger of framework bloat. Android framework is extremely bloated due to years of organic growth and backward compat concerns. E.g. RecyclerView is a monstrosity. It would take *41* pages to print its entire API set. It would be misleading at best to say ListView in Flutter is equivalent to RecyclerView.

That's why we stopped asking ourselves (and you) whether a very specific feature in Android exists in Flutter. Instead, we ask ourselves why we need that feature in the first place and whether there's a pattern which is material design compatible to achieve the same thing and how that can be done in Flutter.

**Comment by sethladd** [comment]
We're collecting ideas and Table of Contents here: https://docs.google.com/document/d/1y8oBQ-K7WFU7DMQS3DnNRqhC5gslGg29cTBV0i2pl-U/edit?ts=59261b06

**Comment by sethladd** [comment]
We largely did this, thanks to @FaisalAbid :)

See https://flutter.io/flutter-for-android/
